core/src/main/java/org/apache/accumulo/core/logging/FateLogger.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.accumulo.core.logging;
+
+import static org.apache.accumulo.fate.FateTxId.formatTid;
+
+import java.io.Serializable;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.function.Function;
+
+import org.apache.accumulo.fate.ReadOnlyRepo;
+import org.apache.accumulo.fate.Repo;
+import org.apache.accumulo.fate.StackOverflowException;
+import org.apache.accumulo.fate.TStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FateLogger {
+  private static final String PREFIX = Logging.PREFIX + "fate.";
+
+  // Logs all mutations to FATEs persistent storage. Enabling this logger could help debug
+  // reproducible problems with FATE transactions.
+  private static final Logger storeLog = LoggerFactory.getLogger(PREFIX + "store");
+
+  public static <T> TStore<T> wrap(TStore<T> store, Function<Repo<T>,String> toLogString) {
```

Comment:
One thing I do not like about this is the function to convert a Repo to a string for logging is not in this class.  However I could not figure out how to make Gson do what I wanted in this class, I had to move the code to master module.  When the code was here Gson kept converting the Repo's to `{}`.

Reply:
One possible solution is to create a simple pojo to be serialized... and put in it whatever you need from the Repo, and serialize the pojo with Gson rather than serialize the Repo. That's what I did for AccumuloMonitorAppender in #1528 rather than try to serialize Log4j's LogEvent type.